### PR TITLE
changelog for 0.17

### PR DESCRIPTION
**Pull request by jasonwilliams**
These are the initial changes for the changelog of 0.17.
Taken from: https://gist.github.com/jasonwilliams/097b42669859b375f3ea1ec748b976ed

There are some "ignored commits" in there which you may want adding back into the changelog.

### CODE REVIEW

**Comment by Razican** [comment]
I would also run `cargo update` in this branch, since dependabot only updates direct dependencies.

**Comment by HalidOdat** [comment]
Added the ignored commits :)

Also Rebased to the latest main (since it was on an old commit) and ran `cargo update`

**Comment by Razican** [comment]
This would need a re-base

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,293 @@
 
 ## What's Changed
 
+# [0.17.0 (2023-05-31)]()
```
Comment (jedel1043) [inline CHANGELOG.md:5]:
```suggestion
# [0.17.0 (2023-06-03)]()
```

**Comment by jedel1043** [comment]
Can you edit the list of ignored commits?

**Comment by jasonwilliams** [comment]
Closing in favour of https://github.com/boa-dev/boa/pull/3101

**Pull request closed without merging**